src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx

```diff
@@ -660,7 +660,36 @@ const getScoreConditions = (items?: Item[], conditions?: Condition[]) =>
     return {
       ...condition,
       payload,
-      itemName: relatedItem ? getEntityKey(relatedItem) : condition.itemName,
+      itemName: relatedItem ? getEntityKey(relatedItem) : scoreName ?? condition.itemName,
+    };
+  });
+
+const findRelatedScore = (itemName: string, scores?: ScoreReport[]) => {
+  if (!scores) return;
+
+  const relatedScore = scores.find((score) => score.id === itemName);
+
+  if (relatedScore) return relatedScore;
+
+  const relatedScoreConditional = scores?.flatMap((score) => score.conditionalLogic);
+
+  return relatedScoreConditional?.find((conditional) => conditional && conditional.id === itemName);
+};
+
+const getSectionConditions = (items?: Item[], conditions?: Condition[], scores?: ScoreReport[]) =>
```

Comment:
could you please use an object for parameters instead of sequence of parameters?

Reply:
done